src/inc/til.h

```diff
@@ -18,6 +18,111 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
 #pragma warning(suppress : 26446) // Suppress bounds.4 check for subscript operator.
         return cont[i];
     }
+
+    // color is a universal integral 8bpp RGBA (0-255) color type implicitly convertible to/from
+    // a number of other color types.
+#pragma warning(push)
+    // we can't depend on GSL here (some libraries use BLOCK_GSL), so we use static_cast for explicit narrowing
+#pragma warning(disable : 26472)
+    struct color
+    {
+        uint8_t r, g, b, a;
+
+        constexpr color() noexcept :
+            r{ 0 },
+            g{ 0 },
+            b{ 0 },
+            a{ 0 } {}
+
+        constexpr color(uint8_t _r, uint8_t _g, uint8_t _b) noexcept :
+            r{ _r },
+            g{ _g },
+            b{ _b },
+            a{ 255 } {}
+
+        constexpr color(uint8_t _r, uint8_t _g, uint8_t _b, uint8_t _a) noexcept :
+            r{ _r },
+            g{ _g },
+            b{ _b },
+            a{ _a } {}
+
+        constexpr color(const color&) = default;
+        constexpr color(color&&) = default;
+        color& operator=(const color&) = default;
+        color& operator=(color&&) = default;
+        ~color() = default;
+
+#ifdef _WINDEF_
+        constexpr color(COLORREF c) :
+            r{ static_cast<uint8_t>(c & 0xFF) },
+            g{ static_cast<uint8_t>((c & 0xFF00) >> 8) },
+            b{ static_cast<uint8_t>((c & 0xFF0000) >> 16) },
+            a{ 255 }
+        {
+        }
+
+        operator COLORREF() const noexcept
+        {
+            return static_cast<COLORREF>(r) | (static_cast<COLORREF>(g) << 8) | (static_cast<COLORREF>(b) << 16);
+        }
+#endif
+
+        // Method Description:
+        // - Converting constructor for any other color structure type containing integral R, G, B, A (case sensitive.)
+        // Notes:
+        // - This and all below conversions make use of std::enable_if and a default parameter to disambiguate themselves.
+        //   enable_if will result in an <error-type> if the constraint within it is not met, which will make this
+        //   template ill-formed. Because SFINAE, ill-formed templated members "disappear" instead of causing an error.
+        template<typename TOther>
+        constexpr color(const TOther& other, std::enable_if_t<std::is_integral_v<decltype(std::declval<TOther>().R)> && std::is_integral_v<decltype(std::declval<TOther>().A)>, int> /*sentinel*/ = 0) :

```

Comment:
Correct me if I'm wrong, but aren't these al actually only checking for `.R` and `.A`? If I had a type with only those members, what would happen?

Reply:
Either the substitution will fail and say there's no constructor, or the constructor will fail and say "there's no member `G`", either of which is fine for us

Reply:
k